civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -6581,3 +6605,117 @@ function civihr_employee_portal_convert_array_option_value($optionGroup, $fieldN
 
   return $data;
 }
+
+/**
+ * Get an array containing existing configuration labels for specified Report.
+ *
+ * @param string $reportName
+ * @return array
+ */
+function civihr_employee_portal_get_report_configuration_list($reportName) {
+  $list = array();
+  $result = db_select('reports_configuration', 'rc')
+    ->fields('rc', array('id', 'label'))
+    ->condition('report_name', $reportName)
+    ->orderBy('label')
+    ->execute();
+  while ($row = $result->fetchAssoc()) {
+    $list[$row['id']] = $row['label'];
+  }
+  return $list;
+}
+
+/**
+ * Print JSON encoded array with Report configuration for given Report name and ID.
+ * 
+ * @param string $reportName
+ * @param int $configId
+ */
+function civihr_employee_portal_get_report_configuration_json($reportName, $configId) {
+  if (empty($reportName) || empty($configId)) {
+    print drupal_json_output(array('status' => 'error'));
+    drupal_exit();
+  }
+  $result = db_select('reports_configuration', 'rc')
+    ->fields('rc', array('json_config'))
+    ->condition('id', (int)$configId)
+    ->condition('report_name', $reportName)
+    ->range(0, 1)
+    ->execute();
+  if ($row = $result->fetchAssoc()) {
+    print drupal_json_output(
+      array(
+        'config' => json_decode($row['json_config']),
+        'status' => 'success',
+      )
+    );
+  } else {
+    print drupal_json_output(array('status' => 'error'));
+  }
+  
+  drupal_exit();
+}
+
+/**
+ * Save Report configuration in database.
+ *
+ * @param string $reportName
+ * @param int $configId
+ */
+function civihr_employee_portal_save_report_configuration_json($reportName, $configId) {
+  if (empty($configId) && empty($_POST['label'])) {
+    print drupal_json_output(array('status' => 'error'));
+    drupal_exit();
+  }
+  $responseData = array(
+    'status' => 'success',
+  );
+  try {
+    if (!empty($configId)) {
+      $result = db_update('reports_configuration')
+        ->condition('report_name', $reportName)
+        ->condition('id', (int)$configId)
+        ->fields(array(
+          'json_config' => json_encode($_POST['json_config'])
+        ))
+        ->execute();
+    } else {
+      $result = db_insert('reports_configuration')
+        ->fields(array(
+          'report_name' => $reportName,
+          'label' => $_POST['label'],
```

Comment:
it is really good to know that drupal methods are just a wrapper for PDO and they use prepared statements by default , thanks for clarifying that :) .